neo/test/iotest/test_medio.py

```diff
@@ -0,0 +1,125 @@
+"""
+Tests of neo.io.medio
+"""
+
+import pathlib
+import unittest
+
+from neo.io.medio import MedIO
+from neo.test.iotest.common_io_test import BaseTestIO
+from neo.test.iotest.tools import get_test_file_full_path
+from neo.io.proxyobjects import (AnalogSignalProxy,
+                SpikeTrainProxy, EventProxy, EpochProxy)
+from neo import (AnalogSignal, SpikeTrain)
+
+import quantities as pq
+import numpy as np
+
+
+# This run standard tests, this is mandatory for all IOs
+class TestMedIO(BaseTestIO, unittest.TestCase, ):
+    ioclass = MedIO
+    entities_to_download = ['med']
+    entities_to_test = [
+        'med/sine_waves.medd',
+        'med/test.medd'
+    ]
+
+    def setUp(self):
+
+        super().setUp()
+        self.dirname = self.get_local_path('med/sine_waves.medd')
+        self.dirname2 = self.get_local_path('med/test.medd')
+        self.password = 'L2_password'
+                
+    def test_read_segment_lazy(self):
+    
+        r = MedIO(self.dirname, self.password)
+        seg = r.read_segment(lazy=False)
+        
+        # There will only be one analogsignal in this reading
+        self.assertEqual(len(seg.analogsignals), 1)
+        # Test that the correct number of samples are read, 5760000 samps for 3 channels
+        self.assertEqual(seg.analogsignals[0].shape[0], 5760000)
+        self.assertEqual(seg.analogsignals[0].shape[1], 3)
+        
+        # Test the first sample value of all 3 channels, which are
+        # known to be [-1, -4, -4]
+        np.testing.assert_array_equal(seg.analogsignals[0][0][:3], [-1, -4, -4])
+        
+        for anasig in seg.analogsignals:
+            assert isinstance(anasig, AnalogSignal)
+            self.assertNotEqual(anasig.size, 0)
+        for st in seg.spiketrains:
+            assert isinstance(st, SpikeTrain)
+            self.assertNotEqual(st.size, 0)
```

Comment:
The `isinstance` checks here are already covered by the neo core library and don't need to be repeated here.
```suggestion
        for anasig in seg.analogsignals:
            self.assertNotEqual(anasig.size, 0)
        for st in seg.spiketrains:
            self.assertNotEqual(st.size, 0)
```